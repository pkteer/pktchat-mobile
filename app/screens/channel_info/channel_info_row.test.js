// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import Preferences from '@mm-redux/constants/preferences';
import {shallowWithIntl} from '@test/intl-test-helper';

import ChannelInfoRow from './channel_info_row';

describe('ChannelInfoRow', () => {
    const baseProps = {
        action: jest.fn(),
        testID: 'test-id',
        defaultMessage: 'default-message',
        detail: 'detail',
        icon: 'icon',
        iconColor: '#ababab',
        image: 1,
        imageTintColor: '#ffffff',
        rightArrow: true,
        textId: 'text-id',
        togglable: false,
        textColor: '#000000',
        theme: Preferences.THEMES.denim,
    };

    test('should match snapshot', () => {
        const wrapper = shallowWithIntl(
            <ChannelInfoRow {...baseProps}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
