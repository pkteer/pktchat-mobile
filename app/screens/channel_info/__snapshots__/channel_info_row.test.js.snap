// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelInfoRow should match snapshot 1`] = `
<TouchableHighlight
  onPress={[MockFunction]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "flexDirection": "row",
        "paddingHorizontal": 15,
      }
    }
    testID="test-id"
  >
    <Image
      source={1}
      style={
        Object {
          "height": 15,
          "tintColor": "#ffffff",
          "width": 15,
        }
      }
    />
    <InjectIntl(FormattedText)
      defaultMessage="default-message"
      id="text-id"
      style={
        Array [
          Object {
            "flex": 1,
            "fontSize": 15,
            "marginLeft": 15,
            "paddingVertical": 15,
          },
          Object {
            "color": "#000000",
          },
        ]
      }
    />
    <Text
      style={
        Object {
          "color": "rgba(63,67,80,0.5)",
          "fontSize": 15,
          "marginHorizontal": 7,
        }
      }
    >
      detail
    </Text>
    <CompassIcon
      name="chevron-right"
      size={24}
      style={
        Object {
          "color": "rgba(63,67,80,0.32)",
          "marginRight": -4,
        }
      }
    />
  </View>
</TouchableHighlight>
`;
